standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/client/HookMetaStoreClientProxy.java

```diff
@@ -0,0 +1,1266 @@
+  public Table getTable(GetTableRequest getTableRequest) throws TException {
+    Table t = getDelegate().getTable(getTableRequest);
+    executePostGetTableHook(t);
+    return HiveMetaStoreClientUtils.deepCopy(
+        FilterUtils.filterTableIfEnabled(isClientFilterEnabled, filterHook, t));
+  }
+
+  private void executePostGetTableHook(Table t) throws MetaException {
+    HiveMetaHook hook = getHook(t);
+    if (hook != null) {
+      hook.postGetTable(t);
+    }
+  }
+
+  @Override
+  public List<Table> getTableObjectsByName(String dbName, List<String> tableNames)
+      throws TException {
+    return getTables(getDefaultCatalog(conf), dbName, tableNames, null);
+  }
+
+  @Override
+  public List<Table> getTableObjectsByName(String catName, String dbName, List<String> tableNames)
+      throws TException {
+    return getTables(catName, dbName, tableNames, null);
+  }
+
+  @Override
+  public List<Table> getTables(String catName, String dbName, List<String> tableNames,
+      GetProjectionsSpec projectionsSpec) throws TException {
+    List<Table> tabs = getDelegate().getTables(catName, dbName, tableNames, projectionsSpec);
+    for (Table tbl : tabs) {
+      executePostGetTableHook(tbl);
+    }
+    return HiveMetaStoreClientUtils.deepCopyTables(
+        FilterUtils.filterTablesIfEnabled(isClientFilterEnabled, filterHook, tabs));
+  }
+
+  @Override
+  public List<String> listTableNamesByFilter(String dbName, String filter, short maxTables)
+      throws TException {
+    return listTableNamesByFilter(getDefaultCatalog(conf), dbName, filter, maxTables);
+  }
+
+  @Override
+  public List<String> listTableNamesByFilter(String catName, String dbName, String filter,
+      int maxTables) throws TException {
+    List<String> tableNames = getDelegate().listTableNamesByFilter(catName, dbName, filter, maxTables);
+    return FilterUtils.filterTableNamesIfEnabled(
+        isClientFilterEnabled, filterHook, catName, dbName, tableNames);
+  }
+
+  @Override
+  public List<String> getTables(String dbname, String tablePattern) throws MetaException {
+    try {
+      return getTables(getDefaultCatalog(conf), dbname, tablePattern);
+    } catch (Exception e) {
+      MetaStoreUtils.throwMetaException(e);
+    }
+    return null;
+  }
+
+  @Override
+  public List<String> getTables(String catName, String dbName, String tablePattern) throws TException {
+    // SG:FIXME, inefficient
+    List<String> tableNames = getDelegate().getTables(catName, dbName, tablePattern);
+    // tables is filtered by FilterHook.
+    List<Table> tables = getTableObjectsByName(catName, dbName, tableNames);
+    List<String> result = new ArrayList<>();
+    for (Table tbl: tables) {
+      result.add(tbl.getTableName());
+    }
+    return result;
+  }
+
+  @Override
+  public List<String> getTables(String dbname, String tablePattern, TableType tableType) throws MetaException {
+    try {
+      return getTables(getDefaultCatalog(conf), dbname, tablePattern, tableType);
```

Comment:
I guess this needs to be `getDelegate().getTables(dbname, tablePattern, tableType)`. That's because `SessionMetaStoreClientProxy` has `getTables(String dbname, String tablePattern, TableType tableType)` but doesn't have `getTables(String catName, String dbname, String tablePattern, TableType tableType)`.